os-boot/Makefile

```diff
@@ -0,0 +1,67 @@
+.PHONY: download_gcc download_linux download_opensbi linux opensbi clean all
+
+all: clean download_gcc download_linux download_opensbi linux opensbi
+
+RISCV_COMPILER_URL ?= https://github.com/riscv-collab/riscv-gnu-toolchain/releases/download/2025.01.20/riscv64-glibc-ubuntu-24.04-gcc-nightly-2025.01.20-nightly.tar.xz
+LINUX_VERSION ?= 6.14.4
+OPENSBI_VERSION ?= 1.6
+
+download_gcc:
+	mkdir -p build
+ifeq (, $(wildcard build/gcc.tar.xz))
+	curl -L $(RISCV_COMPILER_URL) -o build/gcc.tar.xz
+endif
+	curl -L $(RISCV_COMPILER_URL) -o build/gcc.tar.xz
```

Comment:
What's going on here? Looks like it can download it twice.

Btw I found a much nicer way to do downloads with Make after this - see https://github.com/riscv/riscv-isa-manual/pull/1838/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R141

Basically add files with the URLs in them, and then add a Makefile rule that converts the URL files to the download. That way if you edit the URL file it gets redownloaded.

Reply:
It will now download once.

Reply:
> Btw I found a much nicer way to do downloads with Make after this - see https://github.com/riscv/riscv-isa-manual/pull/1838/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R141
> 
> Basically add files with the URLs in them, and then add a Makefile rule that converts the URL files to the download. That way if you edit the URL file it gets redownloaded.

I just cache these things in CI cache so just now I won't do it. just check if these files are exited or not. You can give me more advice.